src/System Application/App/File System/src/Account/FileAccountImpl.Codeunit.al

```diff
@@ -0,0 +1,247 @@
+
+namespace System.FileSystem;
+
+using System.Utilities;
+using System.Text;
+
+codeunit 9451 "File Account Impl."
+{
+    Access = Internal;
+    InherentPermissions = X;
+    InherentEntitlements = X;
+    Permissions = tabledata "File System Connector Logo" = rimd,
+                  tabledata "File Scenario" = imd;
+
+    procedure GetAllAccounts(LoadLogos: Boolean; var TempFileAccount: Record "File Account" temporary)
+    var
+        FileAccounts: Record "File Account";
+        IFileConnector: Interface "File System Connector";
+        Connector: Enum "File System Connector";
+    begin
+        TempFileAccount.Reset();
+        TempFileAccount.DeleteAll();
+
+        foreach Connector in Connector.Ordinals do begin
+            IFileConnector := Connector;
+
+            FileAccounts.DeleteAll();
+            IFileConnector.GetAccounts(FileAccounts);
+
+            if FileAccounts.FindSet() then
+                repeat
+                    TempFileAccount := FileAccounts;
+                    TempFileAccount.Connector := Connector;
+
+                    if LoadLogos then begin
+                        ImportLogo(TempFileAccount, Connector);
+                        ImportLogoBlob(TempFileAccount, Connector);
+                    end;
+
+                    if not TempFileAccount.Insert() then;
+                until FileAccounts.Next() = 0;
+        end;
+
+        // Sort by account name
+        TempFileAccount.SetCurrentKey(Name);
+    end;
+
+    procedure DeleteAccounts(var FileAccountsToDelete: Record "File Account")
+    var
+        CurrentDefaultFileAccount: Record "File Account";
+        ConfirmManagement: Codeunit "Confirm Management";
+        FileScenario: Codeunit "File Scenario";
+        FileConnector: Interface "File System Connector";
+    begin
+        CheckPermissions();
+
+        if not ConfirmManagement.GetResponseOrDefault(ConfirmDeleteQst, true) then
+            exit;
+
+        if not FileAccountsToDelete.FindSet() then
+            exit;
+
+        // Get the current default account to track if it was deleted
+        FileScenario.GetDefaultFileAccount(CurrentDefaultFileAccount);
+
+        // Delete all selected acounts
+        repeat
+            // Check to validate that the connector is still installed
+            // The connector could have been uninstalled by another user/session
+            if IsValidConnector(FileAccountsToDelete.Connector) then begin
+                FileConnector := FileAccountsToDelete.Connector;
+                FileConnector.DeleteAccount(FileAccountsToDelete."Account Id");
+            end;
+        until FileAccountsToDelete.Next() = 0;
+
+        HandleDefaultAccountDeletion(CurrentDefaultFileAccount."Account Id", CurrentDefaultFileAccount.Connector);
+    end;
+
+    local procedure HandleDefaultAccountDeletion(CurrentDefaultAccountId: Guid; Connector: Enum "File System Connector")
```

Comment:
words like Handle, Do, Management, Helper, etc add no information at all so its best to not use them.